versioned_hdf5/replay.py

```diff
@@ -215,35 +234,60 @@ def _recreate_raw_data(
         if fillvalue not in [0, '', b'', None]:
             raise ValueError("Non-default fillvalue not supported for variable length strings")
         fillvalue = None
-    new_raw_data = f['_version_data'][name].create_dataset(
-        '_tmp_raw_data', shape=new_shape, maxshape=(None,)+chunks[1:],
-        chunks=raw_data.chunks, dtype=dtype,
-        compression=raw_data.compression,
-        compression_opts=raw_data.compression_opts,
-        fillvalue=fillvalue)
-    for key, val in raw_data.attrs.items():
-        new_raw_data.attrs[key] = val
-
-    r = raw_data[:]
-    n = np.full(new_raw_data.shape, _get_np_fillvalue(raw_data), dtype=new_raw_data.dtype)
-    raw_data_chunks_map = {}
-    for new_chunk, chunk in zip(chunks.indices(new_shape), chunks_to_keep):
-        # Shrink new_chunk to the size of chunk, in case chunk isn't a full
-        # chunk in one of the dimensions.
-        # TODO: Implement something in ndindex to do this.
-        new_chunk = Tuple(
-            *[Slice(new_chunk.args[i].start,
-                      new_chunk.args[i].start+len(chunk.args[i]))
-              for i in range(len(new_chunk.args))])
-        raw_data_chunks_map[chunk] = new_chunk
-        n[new_chunk.raw] = r[chunk.raw]
-
-    new_raw_data[:] = n
-    if not tmp:
-        del f['_version_data'][name]['raw_data']
-        f['_version_data'][name].move('_tmp_raw_data', 'raw_data')
 
-    return raw_data_chunks_map
+    # Guard against existing _tmp_raw_data
+    _delete_tmp_raw_data(f, name)
+
+    try:
+        new_raw_data = f['_version_data'][name].create_dataset(
```

Comment:
I think the `create_dataset` call should be outside of the `try`, so:
```
new_raw_data = ...create_dataset(...)
try:
    ...
except Exception:
    _delete_tmp_raw_data(f, name)
```
I see that you are checking if `_tmp_raw_data` exists in `_delete_tmp_raw_data` (and we should keep that) but I think this moving `create_dataset` out is cleaner.

Reply:
:100: agree, I've made that change. Thanks for the review!